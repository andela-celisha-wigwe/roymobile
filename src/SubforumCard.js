import React, { Component } from 'react'
import { AppRegistry, StyleSheet, Text, View } from 'react-native'

import styles from './styles'

export default class SubforumCard extends Component {

	cutDescription (text, limit = 100) {
		return text.length > limit ? `${text.substr(0, limit)}...` : text
	}

	render () {
		return (
			<View style={styles.subforumCardView}>
				<View style={styles.subforumCardNameView}>
					<Text style={styles.subforumCardNameText}>
						{ this.props.subforum.name }
					</Text>
				</View>
				<Text  style={styles.subforumCardDescription}>
					{ this.cutDescription(this.props.subforum.description) }
				</Text>
			</View>
		)
	}

}