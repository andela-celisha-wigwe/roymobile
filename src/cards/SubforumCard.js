import React, { Component } from 'react'
import { AppRegistry, StyleSheet, TouchableHighlight, Text, View } from 'react-native'

import styles from './styles'

export default class SubforumCard extends Component {

	cutDescription (text, limit = 100) {
		return text.length > limit ? `${text.substr(0, limit)}...` : text
	}

	constructor (props) {
		super(props);
		this.handleShowSubforum = this.handleShowSubforum.bind(this)
	}

	handleShowSubforum (event) {
		console.log("I am here touch", this.props.subforum)
		this.props.navigator.replace({
			id: "SubforumView",
			params: {
				id: this.props.subforum._id["$oid"],
				subforum: this.props.subforum
			}
		})
	}

	render () {
		return (
			<TouchableHighlight onPress={this.handleShowSubforum}>
				<View style={styles.subforumCardView}>
					<View style={styles.subforumCardNameView}>
						<Text style={styles.subforumCardNameText}>
							{ this.props.subforum.name }
						</Text>
					</View>
					<Text  style={styles.subforumCardDescription}>
						{ this.cutDescription(this.props.subforum.description) }
					</Text>
				</View>
			</TouchableHighlight>
		)
	}

}