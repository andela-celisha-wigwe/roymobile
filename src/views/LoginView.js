'use strick'

import React, { Component } from 'react'

import {
	AppRegistry,
	StyleSheet,
	View,
	Text,
	TextInput,
	TouchableHighlight,
	ScrollView,
	Button
} from 'react-native'

import styles from '../styles/styles'

import UserStore from '../../shared/stores/user_store'
import UserAction from '../../shared/actions/user_action'

export default class extends Component {
	constructor(props) {
		super(props);
		this.state = {
			username: "forumadmin",
			password: "forumadmin"
		}

		this.setUsername = this.setUsername.bind(this)
		this.setPassword = this.setPassword.bind(this)
		this.handleLogin = this.handleLogin.bind(this)
	}

	setUsername (username) {
		this.setState({
			username
		})
	}

	setPassword (password) {
		this.setState({
			password
		})
	}

	handleLogin () {
		const { username, password } = this.state
		UserAction.userLogin({username, password})
	}

	render() {
		return (
			<ScrollView>
				<Text>
					Login
				</Text>

				<TextInput
					style={styles.TextInput}
					onChangeText={this.setUsername}
					value={this.state.username}
				/>

				<TextInput
					style={styles.TextInput}
					onChangeText={this.setPassword}
					value={this.state.password}
				/>

				<Button
					onPress={this.handleLogin}
					title="Login"
					color="#841584"
					accessibilityLabel="Login to application"
				/>
			</ScrollView>
		)
	}
}