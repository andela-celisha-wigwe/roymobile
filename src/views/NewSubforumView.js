'use strict'

import React, { Component } from 'react'

import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableHighlight,
  ScrollView,
  Button,
} from 'react-native';

import SubforumStore from './stores/subforum_store'
import SubforumAction from './actions/subforum_action'

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  contentContainer: {
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

export default class extends Component {
	constructor (props) {
		super(props)
		this.state = {
			name: "",
			description: ""
		}
		this.handleCreateSubforum = this.handleCreateSubforum.bind(this)
	}

	handleCreateSubforum (event) {
		console.log(this.state, "hihihihihi")
		SubforumAction.createSubforum(this.state)
	}

	render () {
		return (
			<ScrollView style={styles.contentContainer}>
				<View style={{flexDirection: 'column', justifyContent: 'space-between'}}>
					<TextInput
						style={{height: 80, width: 300, borderColor: 'gray', borderWidth: 1}}
						onChangeText={(name) => this.setState({name})}
						value={this.state.name}
					/>
					<TextInput
						style={{height: 80, width: 300, borderColor: 'gray', borderWidth: 1}}
						onChangeText={(description) => this.setState({description})}
						value={this.state.description}
					/>
					<Button
						onPress={this.handleCreateSubforum}
						title="Save"
						color="#841584"
						accessibilityLabel="Save subforum"
					/>
				</View>
			</ScrollView>
		)
	}
}