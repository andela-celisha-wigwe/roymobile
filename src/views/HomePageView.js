/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import SubforumCard from './SubforumCard'

import SubforumAction from './actions/subforum_action'
import SubforumStore from './stores/subforum_store'

import SubforumAPI from "../utils/subforum"

import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  ScrollView
} from 'react-native';

export default class extends Component {

  constructor(props) {
    super(props);
    this.state = {
      subforums: [],
      err: {}
    }
    this.onSubforums = this.onSubforums.bind(this)
  }

  componentDidMount() {
    this.unsubscribe = SubforumStore.listen(this.onSubforums)
    SubforumAction.listSubforums()

    // SubforumAPI.all()
    // .then((subforums) => {
    //   this.setState({
    //     subforums
    //   })
    // })
    // .catch((err) => {
    //   this.setState({
    //     err
    //   })
    // })
  }

  onSubforums (subforums) {
    this.setState({
      subforums
    })
  }

  componentWillUnmount () {
    this.unsubscribe()
  }

  render() {
    console.log("I am home view here")
    
    const subforums = this.state.subforums.length > 0 ? this.state.subforums.map((subforum) => {
      return <SubforumCard navigator={this.props.navigator} key={subforum._id["$oid"]} subforum={subforum} />
    }) : null;
    return (
      <ScrollView style={styles.contentContainer}>
        <View style={styles.container}>
          <Text style={styles.welcome}>
            Roy Forums
          </Text>
          { subforums }
        </View>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  contentContainer: {
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

AppRegistry.registerComponent('roymobile', () => roymobile);
