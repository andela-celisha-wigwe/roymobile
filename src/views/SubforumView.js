'use strict'

import React from 'react'
// import {Card, CardActions, CardHeader, CardText} from 'material-ui/Card'

import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TouchableHighlight,
  ScrollView
} from 'react-native';

import SubforumStore from './stores/subforum_store'
import SubforumAction from './actions/subforum_action'
import SubforumCard from './SubforumCard'

// import PostAction from '../actions/post_action'
// import PostStore from '../stores/post_store'
import PostCard from './PostCard'



const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  contentContainer: {
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

export default class extends React.Component {
	constructor (props) {
		super(props)
		this.state = {
			posts: [],
			subforum: this.props.parameters.subforum,
			showForm: true,
			debug: ""
		}
		this.onPosts = this.onPosts.bind(this)
		this.onSubforum = this.onSubforum.bind(this)
	}

	componentDidMount () {
		// this.unsubscribePosts = PostStore.listen(this.onPosts)
		this.unsubscribeSubforum = SubforumStore.listen(this.onSubforum)

		// PostAction.listPosts(this.props.params.id) // how to get the params id.
		// SubforumAction.viewSubforum(this.props.parameters.id)
		this.setState({
			debug: JSON.stringify(Object.keys(this.props))
		})
	}

	componentWillUnmount () {
		// this.unsubscribePosts()
		this.unsubscribeSubforum()
	}

	onSubforum (subforum) {
		this.setState({
			subforum
		})
	}

	onPosts (posts) {
		if (posts.constructor !== Array ) {
			posts = [posts].concat(this.state.posts)
		}
		this.setState({
			posts
		})
	}

	render () {
		const posts = this.state.posts.map((post) =>
			(<PostCard body={post.body} navigator={this.props.navigator} key={post._id["$oid"]} title={post.title} id={post._id["$oid"]} />)
		)
		return (
			<TouchableHighlight>
			<ScrollView style={styles.contentContainer}>
				<View style={styles.container}>
					<Text style={styles.welcome}>
						{this.state.subforum.name}
					</Text>
					<Text style={styles.welcome}>
						{ posts }
					</Text>
				</View>
			</ScrollView>
			</TouchableHighlight>
		)
	}
}