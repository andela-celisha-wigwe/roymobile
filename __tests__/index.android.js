import 'react-native';
import React from 'react';
import Index from '../index.android.js';

import LoginView from '../src/views/LoginView'
import HomePageView from '../src/views/HomePageView'
import SubforumView from '../src/views/SubforumView'
import NewSubforumView from '../src/views/NewSubforumView'

import SubforumCard from '../src/cards/SubforumCard'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Index />
  );
});

it('renders login view correctly', () => {
	const tree = renderer.create(
		<LoginView />
	).toJSON()

	expect(tree).toMatchSnapshot()
});

it('renders home page view correctly', () => {
	const tree = renderer.create(
		<HomePageView />
	).toJSON()

	expect(tree).toMatchSnapshot()
});

it('renders subforum view correctly', () => {

	const subforumMock = { name: "Mock Subforum" }

	const tree = renderer.create(
		<SubforumView parameters={{subforum: subforumMock}} />
	).toJSON()

	expect(tree).toMatchSnapshot()
});

it('renders new subforum view correctly', () => {
	const tree = renderer.create(
		<NewSubforumView />
	).toJSON()

	expect(tree).toMatchSnapshot()
});

it('renders subforum card view with short name correctly', () => {
	const subforum = {
		name: "This is a short name",
		description: "",
	}
	const tree = renderer.create(
		<SubforumCard subforum={ subforum } />
	).toJSON()

	expect(tree).toMatchSnapshot()
});

it('renders subforum card view with long name correctly', () => {

	const subforum = {
		name: ((str) => {
			for (const i = 0; i < 10; i++) {
				str += str
			}
			return str
		})("a"),
		description: "",
	}
	
	const tree = renderer.create(
		<SubforumCard subforum={ subforum } />
	).toJSON()

	expect(tree).toMatchSnapshot()
});
