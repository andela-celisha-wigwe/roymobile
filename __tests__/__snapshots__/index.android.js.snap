exports[`test renders home page view correctly 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#F5FCFF",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        }
      }>
      Roy Forums
    </Text>
  </View>
</ScrollView>
`;

exports[`test renders login view correctly 1`] = `
<ScrollView>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    Login
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "height": 100,
        "width": 400,
      }
    }
    value="forumadmin" />
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "height": 100,
        "width": 400,
      }
    }
    value="forumadmin" />
  <View
    accessibilityComponentType="button"
    accessibilityLabel="Login to application"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Array [
          Object {},
        ]
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#841584",
            },
          ]
        }>
        Login
      </Text>
    </View>
  </View>
</ScrollView>
`;

exports[`test renders new subforum view correctly 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#F5FCFF",
    }
  }>
  <View
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }>
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 80,
          "width": 300,
        }
      }
      value="" />
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 80,
          "width": 300,
        }
      }
      value="" />
    <View
      accessibilityComponentType="button"
      accessibilityLabel="Save subforum"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <View
        style={
          Array [
            Object {},
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#0C42FD",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#841584",
              },
            ]
          }>
          Save
        </Text>
      </View>
    </View>
  </View>
</ScrollView>
`;

exports[`test renders subforum card view with long name correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 100,
        "margin": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0.1875,
        "width": 300,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "#009099",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontWeight": "bold",
          }
        }>
        aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}>
      
    </Text>
  </View>
</View>
`;

exports[`test renders subforum card view with short name correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 100,
        "margin": 0,
        "paddingHorizontal": 0,
        "paddingVertical": 0.1875,
        "width": 300,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "#009099",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffffff",
            "fontWeight": "bold",
          }
        }>
        This is a short name
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={Object {}}>
      
    </Text>
  </View>
</View>
`;

exports[`test renders subforum view correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}>
  <ScrollView
    style={
      Object {
        "backgroundColor": "#F5FCFF",
      }
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5FCFF",
          "flex": 1,
          "justifyContent": "center",
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }>
        Mock Subforum
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        } />
    </View>
  </ScrollView>
</View>
`;
